spec/wizards/schools/register_ect_wizard/ect_spec.rb

```diff
@@ -322,4 +322,150 @@
+        it 'returns nil' do
+          expect(ect.previous_training_programme).to be_nil
+        end
+      end
+    end
+
+    describe '#previous_provider_led?' do
+      it 'returns true when the latest ECTAtSchoolPeriod is provider-led' do
+        FactoryBot.create(:ect_at_school_period, teacher:, programme_type: :provider_led, started_on: Date.new(2024, 1, 1), finished_on: Date.new(2024, 6, 1))
+        expect(ect.previous_provider_led?).to be(true)
+      end
+
+      it 'returns false when the latest ECTAtSchoolPeriod is school-led' do
+        FactoryBot.create(:ect_at_school_period, teacher:, programme_type: :school_led, lead_provider_id: nil, started_on: Date.new(2024, 1, 1), finished_on: Date.new(2024, 6, 1))
+        expect(ect.previous_provider_led?).to be(false)
+      end
+
+      it 'returns nil when there are no ECTAtSchoolPeriods' do
+        expect(ect.previous_provider_led?).to be_nil
+      end
+    end
+
+    describe '#previous_lead_provider_name' do
+      let(:lead_provider) { FactoryBot.create(:lead_provider, name: 'Confirmed LP') }
+
+      before do
+        FactoryBot.create(:training_period,
+                          ect_at_school_period: ect_period,
+                          school_partnership: FactoryBot.create(:school_partnership, lead_provider:),
+                          started_on: Date.new(2024, 1, 1),
+                          finished_on: Date.new(2024, 6, 1))
+      end
+
+      it 'returns the name of the lead provider from the latest training period' do
+        expect(ect.previous_lead_provider_name).to eq('Confirmed LP')
+      end
+    end
+
+    describe '#previous_delivery_partner_name' do
+      let(:delivery_partner) { FactoryBot.create(:delivery_partner, name: 'DP') }
+
+      before do
+        FactoryBot.create(:training_period,
+                          ect_at_school_period: ect_period,
+                          school_partnership: FactoryBot.create(:school_partnership, delivery_partner:),
+                          started_on: Date.new(2024, 1, 1),
+                          finished_on: Date.new(2024, 6, 1))
+      end
+
+      it 'returns the name of the delivery partner from the latest training period' do
+        expect(ect.previous_delivery_partner_name).to eq("DP")
+      end
+    end
+
+    describe '#previous_school_name' do
+      let(:school_1) { FactoryBot.create(:school) }
+      let(:school_2) { FactoryBot.create(:school) }
+
+      before do
+        FactoryBot.create(:gias_school, school: school_1, name: 'Old School')
+        FactoryBot.create(:gias_school, school: school_2, name: 'Recent School')
+
+        FactoryBot.create(:ect_at_school_period, teacher:, school: school_1, started_on: Date.new(2023, 1, 1), finished_on: Date.new(2023, 6, 30))
+        FactoryBot.create(:ect_at_school_period, teacher:, school: school_2, started_on: Date.new(2023, 9, 1), finished_on: nil)
+      end
+
+      it 'returns the name of the most recent school by started_on' do
```

Comment:
Can we put this inside a context so the next context doesn't need to create the ECTAtSchoolPeriod records in the database and then destroy them?. That would speed the build run which tends to take a long time when the project evolves.

Reply:
Good catch!